### PR TITLE
CRM-17847 Fixed custom fields on relationship types for contact subtypes.

### PR DESCRIPTION
**Pull request by mollux**
- [CRM-17847: Fix custom field form notices w\/ relationships between contact sub-types and default types](https://issues.civicrm.org/jira/browse/CRM-17847)

### CODE REVIEW

File: CRM/Contact/BAO/Relationship.php [hunk 1]
```diff
@@ -551,12 +551,13 @@ public static function getContactRelationshipType(
       }
     }
 
-    $contactSubType = array();
     if ($contactId) {
       $contactType = CRM_Contact_BAO_Contact::getContactType($contactId);
       $contactSubType = CRM_Contact_BAO_Contact::getContactSubType($contactId);
     }
 
+    $contactSubType = (array) $contactSubType;
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Relationship.php]: Wouldn't that cause an enotice if $contactId was not set?

Comment (mollux) [inline CRM/Contact/BAO/Relationship.php]: No, as $contactSubType defaults to NULL, an NULL can be casted to an empty array, see https://3v4l.org/nrnGP

**Comment by colemanw** [comment]
@mollux sorry for letting this go stale - appreciate you putting the time into fixing this issue.
Do you mind rebasing this to get it back on track?

**Comment by wdecraene** [comment]
File attached is the patch, based on the latest civicrm-core version.

[CRM-17847.new.patch.txt](https://github.com/civicrm/civicrm-core/files/943205/CRM-17847.new.patch.txt)

**Comment by eileenmcnaughton** [comment]
I just tried to reproduce this problem on dmaster but failed - so I'm guessing this got fixed some other way in the last  8 months - closing

**Pull request closed without merging**